dotnet/src/Microsoft.Agents.AI.Workflows/InProc/InProcStepTracer.cs

```diff
@@ -72,19 +73,19 @@ public override string ToString()
     {
         StringBuilder sb = new();
 
-        if (this.Instantiated.Count != 0)
+        if (this.Instantiated.Keys.Count != 0)
```

Comment:
Keys is unexpectedly expensive. Better to access it once and store it into a local.